pkg/controller/istiocontrolplane/istiocontrolplane_controllerv2.go

```diff
@@ -0,0 +1,97 @@
+	"istio.io/operator/pkg/helmreconciler"
+)
+
+// ReconcileIstioControlPlane reconciles a v1alpha2 IstioControlPlane object
+type ReconcileIstioControlPlaneV2 struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client   client.Client
+	scheme   *runtime.Scheme
+	factory  *helmreconciler.Factory
+	instance runtime.Object
+}
+
+// Reconcile reads that state of the cluster for a v1alpha2 IstioControlPlane object and makes changes based on the state read
+// and what is in the IstioControlPlane.Spec
+func (r *ReconcileIstioControlPlaneV2) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling IstioControlPlane")
+
+	// Fetch the IstioControlPlane instance
+	instance := &v1alpha2.IstioControlPlane{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	deleted := instance.GetDeletionTimestamp() != nil
+	finalizers := instance.GetFinalizers()
+	finalizerIndex := indexOf(finalizers, finalizer)
+
+	if deleted {
+		if finalizerIndex < 0 {
+			reqLogger.Info("IstioControlPlane deleted")
+			return reconcile.Result{}, nil
+		}
+		reqLogger.Info("Deleting IstioControlPlane")
+
+		reconciler, err := r.factory.NewISCPReconciler(instance, r.client, reqLogger)
+		if err == nil {
+			err = reconciler.Delete()
+		} else {
+			reqLogger.Error(err, "failed to create reconciler")
+		}
+		// XXX: for now, nuke the resources, regardless of errors
+		finalizers = append(finalizers[:finalizerIndex], finalizers[finalizerIndex+1:]...)
+		instance.SetFinalizers(finalizers)
+		finalizerError := r.client.Update(context.TODO(), instance)
+		for retryCount := 0; errors.IsConflict(finalizerError) && retryCount < 5; retryCount++ {
+			// workaround for https://github.com/kubernetes/kubernetes/issues/73098 for k8s < 1.14
+			reqLogger.Info("confilict during finalizer removal, retrying")
+			_ = r.client.Get(context.TODO(), request.NamespacedName, instance)
+			finalizers = instance.GetFinalizers()
+			finalizerIndex = indexOf(finalizers, finalizer)
+			finalizers = append(finalizers[:finalizerIndex], finalizers[finalizerIndex+1:]...)
+			instance.SetFinalizers(finalizers)
+			finalizerError = r.client.Update(context.TODO(), instance)
+		}
+		if finalizerError != nil {
+			reqLogger.Error(finalizerError, "error removing finalizer")
+		}
+		return reconcile.Result{}, err
+	} else if finalizerIndex < 0 {
```

Comment:
does this indicate it is an add istio control plane resource?  Can we make info msg more intuitive to users?  I don't think finalizer will mean much to users.

Reply:
I think this is intended to be for debugging - have changed the level to 2.